firefox-ios/Client/Frontend/Home/Homepage Rebuild/Redux/HomepageState.swift

```diff
@@ -6,9 +6,14 @@ import Common
 import Redux
 
 struct HomepageState: ScreenState, Equatable {
+    enum NavigationDestination {
+        case customizeHomepage
+    }
```

Comment:
Menu also navigates to the same homepage settings location; not sure if we want to dispatch a generic action to the BVC state and add a `customizeHomepageSettings` case to `DisplayType` in the BVCState instead of handling it here in Homepage state / Menu state, respectively. Makes sense to dispatch over there since we have a single point in which we present that view. However, it does keep the BVCState from being already bloated. I decided to keep it in `HomepageState` for now unless we decide to refactor for both cases, which can be done in a separate PR. Let me know what you think!

Reply:
Discuss with @OrlaM on this. It seems the way to go is to go through BVC as it should be the single point of knowing how to present the different screens. Will create a separate PR to refactor this.